x-pack/plugins/security_solution/server/lib/detection_engine/rule_types/utils/suppression_utils.ts

```diff
@@ -69,13 +70,13 @@ export const getSuppressionTerms = ({
   alertSuppression: AlertSuppressionCamel | undefined;
 }): SuppressionTerm[] => {
   const suppressedBy = alertSuppression?.groupBy ?? [];
-
   const suppressedProps = pick(fields, suppressedBy) as Record<
     string,
     string[] | number[] | undefined
   >;
+
   const suppressionTerms = suppressedBy.map((field) => {
-    const value = suppressedProps[field] ?? null;
+    const value = get(suppressedProps, field) ?? null;
```

Comment:
@marshallmain @vitaliidm should I use `robustGet` here too? I'm confused by the type casting in this function. The expected return type is `string[] | number[] | null` but the line `Array.isArray(value) ? (sortBy(value) as string[] | number[]) : value;` kind of negates this expectation, right? The unit tests also assert it can be an array or singular value, which I think goes against the stated return type for this function.

Reply:
@dhurley14 , looks like string/number was missed in this type.
Similarly as in 

```ts
export interface SuppressionTerm {
  field: string;
  value: string[] | number[] | null;
}

```

Since fields property is Record<string, unknown>, hence is casting. Fields are typed as `    fields?: Record<string, any>;` in `node_modules/@elastic/elasticsearch/lib/api/typesWithBodyKey.d.ts`.
So nothing much apart basic type check and casting we can do here.
Would you mind to fix type here, by adding `string` and `number`? Or create a separate ticket for this?

Reply:
@dhurley14 , looks like string/number was missed in this type.
Similarly as in 

```ts
export interface SuppressionTerm {
  field: string;
  value: string[] | number[] | null;
}

```

Since fields property is Record<string, unknown>, hence is casting. Fields are typed as `    fields?: Record<string, any>;` in `node_modules/@elastic/elasticsearch/lib/api/typesWithBodyKey.d.ts`.
So nothing much apart basic type check and casting we can do here.
Would you mind to fix type here, by adding `string` and `number`? Or create a separate ticket for this?

Reply:
@dhurley14 , looks like string/number was missed in this type.
Similarly as in 

```ts
export interface SuppressionTerm {
  field: string;
  value: string[] | number[] | null;
}

```

Since fields property is Record<string, unknown>, hence is casting. Fields are typed as `    fields?: Record<string, any>;` in `node_modules/@elastic/elasticsearch/lib/api/typesWithBodyKey.d.ts`.
So nothing much apart basic type check and casting we can do here.
Would you mind to fix type here, by adding `string` and `number`? Or create a separate ticket for this?

Reply:
Okay great I will fix it here 👍